lib/Dialect/FIRRTL/Transforms/InferWidths.cpp

```diff
@@ -1489,17 +1495,27 @@ LogicalResult InferenceMapping::mapOperation(Operation *op) {
         // actually want is for all data ports to share the same variable. To do
         // this, we find the first data port declared, and use that port's vars
         // for all the other ports.
-        unsigned firstFieldIndex = dataFieldIndices(op.getPortKind(0))[0];
-        FieldRef firstData(op.getResult(0), op.getPortType(0)
-                                                .getPassiveType()
-                                                .template cast<BundleType>()
-                                                .getFieldID(firstFieldIndex));
+        unsigned firstFieldIndex =
+            dataFieldIndices(op.getPortKind(nonDebugPort))[0];
+        FieldRef firstData(op.getResult(nonDebugPort),
+                           op.getPortType(nonDebugPort)
+                               .getPassiveType()
+                               .template cast<BundleType>()
+                               .getFieldID(firstFieldIndex));
         LLVM_DEBUG(llvm::dbgs() << "Adjusting memory port variables:\n");
 
         // Reuse data port variables.
         auto dataType = op.getDataType();
         for (unsigned i = 0, e = op.getResults().size(); i < e; ++i) {
           auto result = op.getResult(i);
+          if (result.getType().cast<FIRRTLType>().isa<RefType>()) {
+            // Debug ports are firrtl.ref<vector<data-type>, depth>
```

Comment:
Should the depth go with the vector?
```suggestion
            // Debug ports are firrtl.ref<vector<data-type, depth>>
```